source/adapters/cuda/common.cpp

```diff
@@ -111,7 +111,7 @@ void detail::ur::assertion(bool Condition, const char *Message) {
 }
 
 void detail::ur::cuPrint(const char *Message) {
-  std::cerr << "ur_print: " << Message << std::endl;
+  std::fprintf(stderr, "ur_print: %s\n", Message);
```

Comment:
Why was this change necessary?

Reply:
`cerr` was catching a weird seg fault that was avoided with `fprintf`

Reply:
It worries me a bit that we never found out what exactly was causing the segfault. It might be a segfault / undefined behavior in some other part of the adapter that for some reason tends to crash inside std::cerr. I think we should investigate this further. Maybe running a sanitizer or valgrind might help.

However, since it seems to pass with the change to fprintf, maybe this PR can still be merged as long as we look into it afterwards.

Reply:
> However, since it seems to pass with the change to fprintf, maybe this PR can still be merged as long as we look into it afterwards.

Agreed, open an issue to investigate this later.

Reply:
The issue here was that the changes from this [PR](https://github.com/oneapi-src/unified-runtime/pull/1016) was not included due to rebasing on adapters revert commit, so the deepbind loading of the loaders was still there, and that was causing this issue. once I have included the revert of the revert commit (the commit for this [PR](https://github.com/oneapi-src/unified-runtime/pull/1016)) and the deepbind removal was included. This problem have been fixed. So, it should be fixed now.